### PR TITLE
migrate to Navigator.resetRideSession with callback

### PR DESCRIPTION
**Pull request by LukasPaczos**
### Description

Removes usage of the deprecated `resetRideSession` and exposes a new one with a callback.

### CODE REVIEW

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt [hunk 1]
```diff
@@ -638,11 +639,38 @@ class MapboxNavigation @VisibleForTesting internal constructor(
...
+
+    /**
+     * Reset the session with the same configuration. The location becomes unknown,
+     * but the [NavigationOptions] stay the same.
+     *
+     * Call this function before significant change of location, e.g. when restarting
+     * navigation simulation, or before resetting location to not real (simulated)
+     * position without recreation of [MapboxNavigation].
+     */
+    fun resetTripSession(callback: ResetTripSessionCallback) {
```
Comment (dzinad) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt:663]: current.txt

File: libnavigation-core/src/test/java/com/mapbox/navigation/core/MapboxNavigationTest.kt [hunk 2]
```diff
@@ -770,7 +770,16 @@ internal class MapboxNavigationTest : MapboxNavigationBaseTest() {
...
+        coVerify { navigator.resetRideSession() }
+    }
+
+    @Test
+    fun `resetTripSession should reset the navigator and call back`() {
+        createMapboxNavigation()
+        val callback = mockk<ResetTripSessionCallback>(relaxUnitFun = true)
+        mapboxNavigation.resetTripSession(callback)
+
+        verify { callback.onTripSessionReset() }
```
Comment (dzinad) [inline libnavigation-core/src/test/java/com/mapbox/navigation/core/MapboxNavigationTest.kt:782]: Does it work without coVerify?

File: instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/MapboxHistoryTest.kt [hunk 3]
```diff
@@ -354,6 +356,19 @@ class MapboxHistoryTest : BaseTest<EmptyTestActivity>(EmptyTestActivity::class.j
         assertEquals(thirdJson, thirdFile.findCustomEvent().properties)
     }
 
+    @Test
+    fun trip_session_resets_successfully() {
```
Comment (dzinad) [inline instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/MapboxHistoryTest.kt:360]: Why is it in `MapboxHistoryTest`? Is this related?

Comment (dzinad) [inline instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/MapboxHistoryTest.kt:360]: NIT:
```
    @Test
    fun trip_session_resets_successfully() = sdkTest(timeout = 5000) {
        mapboxNavigation.resetTripSessionAndWaitForResult()
    }
```

And:
```
suspend fun MapboxNavigation.resetTripSessionAndWaitForResult() = suspendCoroutine<Unit> { cont ->
    val callback = ResetTripSessionCallback {
        cont.resume(Unit)
    }
    resetTripSession(callback)
}
```

The second method can then be later reused.

Comment (LukasPaczos) [reply instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/MapboxHistoryTest.kt:360]: We typically use this function with history and replayer, I can create a new test class though, I was on the fence about that.

**Comment by LukasPaczos** [comment]
Ready for another round.

**Review by dzinad — APPROVED** [review]
